sql/core/src/main/scala/org/apache/spark/sql/DataFrameReader.scala

```diff
@@ -245,12 +245,19 @@ class DataFrameReader private[sql](sparkSession: SparkSession) extends Logging {
         "read files of Hive data source directly.")
     }
 
+    if (extraOptions.contains("path") && paths.nonEmpty) {
```

Comment:
how about we forbid `paths` option as well for consistency?

Reply:
Then we can make sure `path` and `paths` options wouldn't co-exist for DS v2, and you can make the change in https://github.com/apache/spark/pull/29328/files#r468295791

Reply:
I added the check for `paths` option.

Should we handle the case where `path` and `paths` options are both defined?